component/prometheus/relabel/relabel.go

```diff
@@ -154,12 +159,26 @@ func (c *Component) clearCache() {
 	c.cacheMut.Lock()
 	defer c.cacheMut.Unlock()
 
-	c.cache = make(map[uint64]*prometheus.FlowMetric)
+	c.cache = make(map[uint64]*labelAndID)
 }
 
 func (c *Component) addToCache(originalID uint64, fm *prometheus.FlowMetric) {
 	c.cacheMut.Lock()
 	defer c.cacheMut.Unlock()
 
-	c.cache[originalID] = fm
+	if fm == nil {
+		c.cache[originalID] = nil
+		return
```

Comment:
Should this delete the entry instead of setting it to nil?

Reply:
This handles the case of dropping a metric entirely.

Reply:
Ah, got it 👍